File: solutions/Leetcode_633/leetcode_633.py
import math


class Solution1:
    def judge_sqaure_sum(c: int) -> bool:
        m = math.floor(math.sqrt(c))
        n = 0
        while m >= n:
            current = m**2 + n ** 2
            if current == c:
                return True
            if current < c:
                n += 1
            else:
                m -= 1
        return False

File: solutions/Leetcode_633/leetcode_633.cpp
#include <cmath>

class Solution1
{
public:
    bool judgeSquareSum(int c)
    {
        int m = std::sqrt(c), n = 0;
        while (m >= n)
        {
            int diff = c - m * m;
            if (diff == n * n)
                return true;
            if (diff > n * n)
                n++;
            else
                m--;
        }
        return false;
    }
};
